weblate/vcs/git.py

```diff
@@ -279,6 +280,33 @@ def configure_remote(self, pull_url, push_url, branch):
         )
         self.branch = branch
 
+    def api_url(self):
+        return self.component.repo.replace(".git", "").replace(
+            "github.com", "api.github.com/repos"
+        )
+
+    def configure_fork_remote(self, pull_url, remote_name):
+        """Configure fork remote repository."""
+        self.config_update(
+            # Pull url
+            ('remote "{}"'.format(remote_name), "url", pull_url),
+            # Push url
+            ('remote "{}"'.format(remote_name), "pushurl", pull_url),
+        )
+
+    def fork(self):
+        fork_url = "{}/forks".format(self.api_url())
+        r = requests.post(
+            fork_url,
+            headers={
+                "Accept": "application/vnd.github.v3+json",
+                "Authorization": "token {}".format(settings.GITHUB_TOKEN),
+            },
+            data={},
+        )
+        response = r.json()
+        self.configure_fork_remote(response["clone_url"], self.get_username())
```

Comment:
This all code is GitHub specific, it should go to the GithubRepository class.

Reply:
Makes sense.